### PR TITLE
Add TFLite Micro examples

### PR DESCRIPTION
**Pull request by sanketwadekar**
## Description of Change
Added TFLite Micro examples.

Depends on https://github.com/espressif/esp32-arduino-lib-builder/pull/128

## Tests scenarios
I have tested these examples on ESP-EYE.

cc:@vikramdattu

### CODE REVIEW

File: libraries/TFLiteMicro/src/utility.h [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/*
+ * SPDX-FileCopyrightText: 2022 Espressif Systems (Shanghai) CO LTD
```
Comment (vikramdattu) [inline libraries/TFLiteMicro/src/utility.h:2]: Please update the license year to 2023 wherever applicable.

File: libraries/TFLiteMicro/src/utility.h [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#pragma once
+#include <utility>
+/*
+The example sketches include "tensorflow/lite/micro/micro_interpreter.h" which internally include "utility.h" header file when compiling examples for Arduino (when -DARDUINO flag is passed to the compiler). This header file does not exist in esp32-arduino core. Hence, keeping this file here as a workaround and including an alternate header file.
+*/
```
Comment (vikramdattu) [inline libraries/TFLiteMicro/src/utility.h:10]: add a new line, do check for other files as well.

File: libraries/TFLiteMicro/examples/micro_speech/yes_micro_features_data.cpp [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    -128, -128, -128, -128, -128, -128, -128, -128, -128, -128, -128, -128,
+    -128, -128, -128, -128, -128, -128, -128, -128, -128, -128, -128, -128,
+    -128, -128, -128, -128, -128, -128, -128, -128, -128, -128, -128, -128,
+    -128, -128, -128, -128, -128, -128, -128, -128, -128, -128, -128, -128,
+    -128, -128, -128, -128, -128, -128, -128, -128, -128, -128, -128, -128,
+    -128, -128, -128, -128, -128, -128, -128, -128, -128, -128, -128, -128,
+    -128, -128, -128, -128, -128, -128, -128, -128, -128, -128, -128, -128,
+    -128, -128, -128, -128, -128, -128, -128, -128, -128, -128, -128, -128,
+    -128, -128, -128, -128, -128, -128, -128, -128, -128, -128, -128, -128,
+    -128, -128, -128, -128,
```
Comment (vikramdattu) [inline libraries/TFLiteMicro/examples/micro_speech/yes_micro_features_data.cpp:187]: Although these files are present in original sources, these are not used here. Please remove these.

File: libraries/TFLiteMicro/examples/micro_speech/no_micro_features_data.cpp [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "no_micro_features_data.h"
+
+// Golden test values for the expected spectrogram from a "no" sample file
```
Comment (vikramdattu) [inline libraries/TFLiteMicro/examples/micro_speech/no_micro_features_data.cpp:18]: Please remove this file as well. Not used anymore.

**Review by vikramdattu — COMMENTED** [review]
@sanketwadekar thanks for adding the Arduino library. Couple of comments, and it should be good to go.

**Comment by P-R-O-C-H-Y** [comment]
@sanketwadekar CI is failing:
```
arduino-esp32/libraries/TFLiteMicro/examples/hello_world/hello_world.ino:17:10: fatal error: tensorflow/lite/micro/micro_mutable_op_resolver.h: No such file or directory
 #include "tensorflow/lite/micro/micro_mutable_op_resolver.h"
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by sanketwadekar** [comment]
> @sanketwadekar CI is failing:
> 
> ```
> arduino-esp32/libraries/TFLiteMicro/examples/hello_world/hello_world.ino:17:10: fatal error: tensorflow/lite/micro/micro_mutable_op_resolver.h: No such file or directory
>  #include "tensorflow/lite/micro/micro_mutable_op_resolver.h"
>           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> ```

This file is not yet present in esp32-arduino. It will be copied by the lib-builder.

**Comment by me-no-dev** [comment]
@sanketwadekar please pull the latest changes from the target branch. TFLite is merged and we can see if CI will pass

**Comment by vikramdattu** [comment]
> @sanketwadekar please pull the latest changes from the target branch. TFLite is merged and we can see if CI will pass

@me-no-dev this will require lib-builder release to happen first, right?

**Comment by me-no-dev** [comment]
> @me-no-dev this will require lib-builder release to happen first, right?

Starting with IDF-v5.1 we build and get the libs in a different way. I have already added TFLite to the libs for 5.1 and have updated the branch. @sanketwadekar needs to pull all changes from that branch and that will trigger CI to re-run the tests with TFLite available.

**Review by lucasssvaz — APPROVED** [review]
Works as expected on ESP32-DevKitC

**Comment by me-no-dev** [comment]
Closing in favor of: https://github.com/espressif/arduino-esp32/pull/8717

**Pull request closed without merging**